js/webtorrent/entry.js

```diff
@@ -1,166 +1,155 @@
-      var ix = Number(state.magnetInfo.ix)
-      content = (
-        <div className='content'>
-          <iframe src={serverURL + '/' + ix} sandbox='' seamless='seamless' />
-        </div>
-      )
-    } else {
-      content = (
-        <div className='content'>
-          Loading...
-        </div>
-      )
-    }
-  } else {
-    var fileElems = torrent.files.map((file, i) => {
-      return (
-        <li>
-          <a href={id + '&ix=' + i}>{file.name}</a>
-        </li>
-      )
-    })
 
-    var fileList
-    if (fileElems.length === 0) {
-      fileList = (<div>Loading...</div>)
-    } else {
-      fileList = (
-        <ul className='files'>
-          {fileElems}
-        </ul>
-      )
+  render () {
+    const ix = state.parsedTorrent && state.parsedTorrent.ix // Selected file index
+
+    const legalNotice = state.torrent == null
+      ? <div className='legalNotice' data-l10n-id='legalNotice' />
+      : <div className='legalNotice' data-l10n-id='poweredByWebTorrent' />
+
+    let fileContent
+    if (state.torrent && ix) {
+      fileContent = state.server != null
+        ? <iframe src={state.server + '/' + ix} sandbox='' seamless='seamless' />
+        : <div>Loading...</div>
     }
 
-    content = (
-      <div className='content'>
-        <h3>Files</h3>
-        {fileList}
+    return <div className='siteDetailsPage'>
+      <div className='siteDetailsPageHeader'>
+        <div data-l10n-id='webtorrentPage' className='sectionTitle' />
+        <div className='sectionTitle'>: {this.props.name}</div>
+        <div className='headerActions'>
+          <TorrentStats torrent={state.torrent} errorMessage={state.errorMessage} />
+          <Button
+            l10nId='startDownload'
+            className='whiteButton startDownload'
+            disabled={!!state.torrent}
+            onClick={this.start} />
+          <Button
+            l10nId='saveTorrentFile'
+            className='whiteButton saveTorrentFile'
+            onClick={this.saveTorrentFile} />
+        </div>
       </div>
-    )
-  }
 
-  var elem = (
-    <div>
-      <h1>{title}</h1>
-      <div className='status-bar'>
-        {status}
-        <span className='button'>
-          <span className='fa fa-magnet' />
-          <span>Copy magnet link</span>
-        </span>
-        <span className='button'>
-          <span className='fa fa-file-o' />
-          <span>Save torrent file</span>
-        </span>
+      <div className='siteDetailsPageContent'>
+        <div className='fileContent'>
+          {fileContent}
+        </div>
+        <TorrentFileList
+          files={state.torrent && state.torrent.files}
+          stateOwner={this}
+          torrentID={state.torrentID} />
+        {legalNotice}
       </div>
-      {content}
-      <div className='error'>{state.errorMessage}</div>
     </div>
-  )
-  ReactDOM.render(elem, document.querySelector('#appContainer'))
+  }
 }
 
+// TODO(feross): Test how this behaves with different file types
```

Comment:
Yeah, I was experimenting with this last night, and ended up removing `resizeToContent` entirely.

Sizing an `<iframe>` to its content is pretty janky in the best situations, and doesn't work at all in cases like this where the outer URL and inner URL have different schemes. (`chrome-extension` vs `http`)